flocker/control/test/test_model.py

```diff
@@ -205,6 +249,24 @@ class NodeStateTests(SynchronousTestCase):
     """
     Tests for ``NodeState``.
     """
+    def test_no_uuid(self):
+        """
+        If no UUID is given, a UUID is generated from the hostname.
+
+        This is done for backwards compatibility with existing tests, and
+        should be removed enveutally.
+        """
+        node = NodeState(hostname=u'1.2.3.4')
+        self.assertIsInstance(node.uuid, UUID)
+
+    def test_uuid(self):
+        """
+        ``NodeState`` can be created with a UUID.
+        """
+        uuid = uuid4()
+        node = NodeState(hostname=u'1.2.3.4', uuid=uuid)
+        self.assertEqual(node.uuid, uuid)
+
```

Comment:
:question: 

It would be nice if the two variations of these two tests, one for `Node` and one for `NodeState`, were shared instead of duplicated.  Less code to maintain, less to clean up later, etc.